src/main/java/dev/ferrand/chunky/octree/implementations/PackedOctree.java

```diff
@@ -231,34 +226,16 @@ private boolean nodeEquals(int firstNodeIndex, int secondNodeIndex) {
     return treeData[firstNodeIndex] == treeData[secondNodeIndex]; // compare types
   }
 
-  /**
-   * Compare two nodes
-   *
-   * @param firstNodeIndex The index of the first node
-   * @param secondNode     The second node (most likely outside of tree)
-   * @return true id the nodes compare equals, false otherwise
-   */
-  private boolean nodeEquals(int firstNodeIndex, Node secondNode) {
-    boolean firstIsBranch = treeData[firstNodeIndex] > 0;
-    boolean secondIsBranch = (secondNode.type == BRANCH_NODE);
-    return ((firstIsBranch && secondIsBranch) || -treeData[firstNodeIndex] == secondNode.type); // compare types (don't forget that in the tree the negation of the type is stored)
-  }
-
   @Override
   public void set(int type, int x, int y, int z) {
-    set(new Node(type), x, y, z);
-  }
-
-  @Override
-  public void set(Octree.Node data, int x, int y, int z) {
     int[] parents = new int[depth]; // better to put as a field to preventallocation at each invocation?
     int nodeIndex = 0;
     int parentLevel = depth - 1;
     int position = 0;
     for(int i = depth - 1; i >= 0; --i) {
       parents[i] = nodeIndex;
 
-      if(nodeEquals(nodeIndex, data)) {
+      if(nodeEquals(nodeIndex, getNodeIndex(x, y, z))) {
```

Comment:
This is wrong, we should compare the current node against the type given as parameter, aka `if(-treeData[nodeIndex] == type)`